CMake Error at tests/unit/CMakeLists.txt:356...Unknown arguments specified

Hello, my full cmake configuring output like this:
```
CMake version: 3.30.0
Loading USER_CONFIG = /home/alex/workspace/MFEM/config/user.cmake (NOTFOUND)
USE_XSDK_DEFAULTS = 'OFF'
Using CUDA architecture: sm_60
Floating-point precision: MFEM_PRECISION = double
MFEM: using package OPENMP
MFEM: using package HYPRE
MFEM: using package LAPACK
MFEM: using package BLAS
MFEM: using package METIS
MFEM: using package POSIXCLOCKS
MFEM: using package ZLIB
MFEM: using package CEED
MFEM: using package CUSPARSE
MFEM: using package CUBLAS
MFEM: using package MPI_CXX
MFEM shared library: BUILD_SHARED_LIBS = OFF
MFEM build type: CMAKE_BUILD_TYPE = Release
MFEM version: v4.7.1
MFEM git string: heads/comments_for_understanding-0-g648dbc47bb8c637a2cb5188da51fb0a035d5ca48-dirty
TPL_INCLUDE_DIRS = sr/include/hypre;/usr/include;/usr/local/include;/opt/openmpi/4.1.6/include
Writing substitute header --> "mfem.h"
Writing substitute header --> "mfem-performance.h"
CMake Error at tests/unit/CMakeLists.txt:356 (if):
  if given arguments:

    "NOT" "(" "(" "sedov" "STREQUAL" "sedov" ")" "AND" "MFEM_DEBUG" "AND" "(" "HYPRE_USING_CUDA" "OR" "HYPRE_USING_HIP" ")" "AND" "(" "LESS" "23100" ")" ")"

  Unknown arguments specified
Call Stack (most recent call first):
  tests/unit/CMakeLists.txt:380 (add_parallel_miniapp_test)


Configuring incomplete, errors occurred!
```
and 

```
USE_CEED
USE_CUDA
USE_LAPACK
USE_METIS
USE_MPI
USE_OPENMP
USE_ZLIB
```

are enabled.

How to solve this problem?

> I added a fix in [#4737](https://github.com/mfem/mfem/pull/4737) to make sure CMake will re-detect hypre if somehow the cmake cache ends up corrupted like this.

After pulling latest master (2fbc743d47ff05de1485415f78c803d03c594051), the configuration problem gone, but got another compilation error:

```
/home/alex/workspace/MFEM/fem/ceed/solvers/full-assembly.cpp(203): error: argument of type "CeedInt (*)[3]" is incompatible with parameter of type "CeedInt *"
     ierr = CeedElemRestrictionGetELayout(rstr_q, &layout); do { if ((ierr)) { const char * errmsg; CeedGetErrorMessage(internal::ceed, &errmsg); { std::ostringstream mfemMsgStream; mfemMsgStream << std::setprecision(16); mfemMsgStream << std::setiosflags(std::ios_base::scientific); mfemMsgStream << "MFEM abort: " << errmsg << "\n ... in function: " << __PRETTY_FUNCTION__ << "\n ... in file: " << "/home/alex/workspace/MFEM/fem/ceed/solvers/full-assembly.cpp" << ':' << 203 << '\n'; mfem::mfem_error(mfemMsgStream.str().c_str()); }; } } while(0);;
                                                  ^

1 error detected in the compilation of "/home/alex/workspace/MFEM/fem/ceed/solvers/full-assembly.cpp".
make[2]: *** [CMakeFiles/mfem.dir/build.make:1176: CMakeFiles/mfem.dir/fem/ceed/solvers/full-assembly.cpp.o] Error 2
make[2]: *** Waiting for unfinished jobs....
/home/alex/workspace/MFEM/fem/ceed/solvers/solvers-atpmg.cpp(123): error: argument of type "CeedInt (*)[3]" is incompatible with parameter of type "CeedInt *"
     ierr = CeedElemRestrictionGetELayout(er_in, &in_layout); do { if ((ierr)) { const char * errmsg; CeedGetErrorMessage(internal::ceed, &errmsg); { std::ostringstream mfemMsgStream; mfemMsgStream << std::setprecision(16); mfemMsgStream << std::setiosflags(std::ios_base::scientific); mfemMsgStream << "MFEM abort: " << errmsg << "\n ... in function: " << __PRETTY_FUNCTION__ << "\n ... in file: " << "/home/alex/workspace/MFEM/fem/ceed/solvers/solvers-atpmg.cpp" << ':' << 123 << '\n'; mfem::mfem_error(mfemMsgStream.str().c_str()); }; } } while(0);;
                                                 ^

/home/alex/workspace/MFEM/fem/ceed/solvers/solvers-atpmg.cpp(668): warning #167-D: argument of type "char **" is incompatible with parameter of type "const char **"
        ierr = CeedQFunctionFieldGetName(inputqfields[i], &fieldname); do { if ((ierr)) { const char * errmsg; CeedGetErrorMessage(internal::ceed, &errmsg); { std::ostringstream mfemMsgStream; mfemMsgStream << std::setprecision(16); mfemMsgStream << std::setiosflags(std::ios_base::scientific); mfemMsgStream << "MFEM abort: " << errmsg << "\n ... in function: " << __PRETTY_FUNCTION__ << "\n ... in file: " << "/home/alex/workspace/MFEM/fem/ceed/solvers/solvers-atpmg.cpp" << ':' << 668 << '\n'; mfem::mfem_error(mfemMsgStream.str().c_str()); }; } } while(0);;
                                                          ^

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"

/home/alex/workspace/MFEM/fem/ceed/solvers/solvers-atpmg.cpp(683): warning #167-D: argument of type "char **" is incompatible with parameter of type "const char **"
        ierr = CeedQFunctionFieldGetName(outputqfields[i], &fieldname); do { if ((ierr)) { const char * errmsg; CeedGetErrorMessage(internal::ceed, &errmsg); { std::ostringstream mfemMsgStream; mfemMsgStream << std::setprecision(16); mfemMsgStream << std::setiosflags(std::ios_base::scientific); mfemMsgStream << "MFEM abort: " << errmsg << "\n ... in function: " << __PRETTY_FUNCTION__ << "\n ... in file: " << "/home/alex/workspace/MFEM/fem/ceed/solvers/solvers-atpmg.cpp" << ':' << 683 << '\n'; mfem::mfem_error(mfemMsgStream.str().c_str()); }; } } while(0);;
                                                           ^

1 error detected in the compilation of "/home/alex/workspace/MFEM/fem/ceed/solvers/solvers-atpmg.cpp".
make[2]: *** [CMakeFiles/mfem.dir/build.make:1183: CMakeFiles/mfem.dir/fem/ceed/solvers/solvers-atpmg.cpp.o] Error 2
make[1]: *** [CMakeFiles/Makefile2:946: CMakeFiles/mfem.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

--- a/config/cmake/modules/FindHYPRE.cmake
+++ b/config/cmake/modules/FindHYPRE.cmake
@@ -25,7 +25,12 @@ if (HYPRE_FOUND)
     find_package(rocsparse REQUIRED)
     find_package(rocrand REQUIRED)
   endif()
-  return()
+  if (HYPRE_LIBRARIES AND HYPRE_INCLUDE_DIRS AND HYPRE_VERSION)
+    find_package_handle_standard_args(HYPRE
+      REQUIRED_VARS HYPRE_LIBRARIES HYPRE_INCLUDE_DIRS HYPRE_VERSION
+    )
+    return()
+  endif()
 endif()
 
 include(MfemCmakeUtilities)
@@ -95,3 +100,7 @@ if (HYPRE_FOUND AND HYPRE_USING_HIP)
       "HYPRE libraries + dependencies." FORCE)
   message(STATUS "Updated HYPRE_LIBRARIES: ${HYPRE_LIBRARIES}")
 endif()
+
+find_package_handle_standard_args(HYPRE
+  REQUIRED_VARS HYPRE_LIBRARIES HYPRE_INCLUDE_DIRS HYPRE_VERSION
+)

